site/_docs/java-tools.md

```diff
@@ -11,43 +11,136 @@ supports both the local file system and HDFS.
 
 The subcommands for the tools are:
 
-  * meta - print the metadata of an ORC file
-  * data - print the data of an ORC file
-  * scan (since ORC 1.3) - scan the data for benchmarking
   * convert (since ORC 1.4) - convert JSON files to ORC
+  * count (since ORC 1.6) - recursively find *.orc and print the number of rows
+  * data - print the data of an ORC file
   * json-schema (since ORC 1.4) - determine the schema of JSON documents
+  * key (since ORC 1.5) - print information about the encryption keys
+  * meta - print the metadata of an ORC file
+  * scan (since ORC 1.3) - scan the data for benchmarking
+  * version (since ORC 1.6) - print the version of this ORC tool
 
 The command line looks like:
 
 ~~~ shell
 % java -jar orc-tools-X.Y.Z-uber.jar <sub-command> <args>
 ~~~
 
+## Java Convert
+
+The convert command reads several JSON files and converts them into a
+single ORC file.
+
+-e,--escape `<escape>`
+  : Sets CSV escape character
+
+-h,--help
+  : Print help
+
+-H,--header `<header>`
+  : Sets CSV header lines
+
+-n,--null `<null>`
+  : Sets CSV null string
+
+-o,--output `<filename>`
+  : Sets the output ORC filename, which defaults to output.orc
+
+-O,--overwrite
+  : If the file already exists, it will be overwritten
+
+-q,--quote `<quote>`
+  : Sets CSV quote character
+
+-s,--schema `<schema>`
+  : Sets the schema for the ORC file. By default, the schema is automatically discovered.
+
+-S,--separator `<separator>`
+  : Sets CSV separator character
+
+-t,--timestampformat `<timestampformat>`
+  : Sets timestamp Format
+  
+The automatic JSON schema discovery is equivalent to the json-schema tool
+below.
+
+## Java Count
+
+The count command recursively find *.orc and print the number of rows. The parameter value can be a space separated file paths string.
+
+## Java Data
+
+The data command prints the data in an ORC file as a JSON document. Each
+record is printed as a JSON object on a line. Each record is annotated with
+the fieldnames and a JSON representation that depends on the field's type.
+
+-h,--help
```

Comment:
ditto. `--` becomes `-` in HTML generation.

Reply:
Fix it in da9740a